example/v1/0000_50_stabletype-techpreview.crd.yaml

```diff
@@ -56,16 +83,26 @@ spec:
                     - StableValue
                     - TechPreviewOnlyValue
                     type: string
-                required:
-                - type
                 type: object
+              immutableField:
+                description: immutableField is a field that is immutable once the
+                  object has been created. It is required at all times.
+                type: string
+                x-kubernetes-validations:
+                - rule: self == oldSelf
```

Comment:
i saw this `self == oldSelf` in the code comments as well, it's slightly non-obvious to me what it is doing. is this a standard part of these definitions/vallidations?

Reply:
@JoelSpeed shared this link with me, it helped my understanding: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules